components/Dialogs/src/MDCAlertController.m

```diff
@@ -186,6 +189,48 @@ - (void)addAction:(MDCAlertAction *)action {
   [self.view setNeedsLayout];
 }
 
+- (BOOL)mdc_adjustsFontForContentSizeCategory {
+  return _mdc_adjustsFontForContentSizeCategory;
+}
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
+  _mdc_adjustsFontForContentSizeCategory = adjusts;
+
+  // Update any action buttons to match the alert controller.
+  for (MDCButton *actionButton in self.actionButtons) {
+    actionButton.mdc_adjustsFontForContentSizeCategory = adjusts;
+  }
+
+  if (_mdc_adjustsFontForContentSizeCategory) {
+    [self updateFontsForDynamicType];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(contentSizeCategoryDidChange:)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
+  } else {
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:UIContentSizeCategoryDidChangeNotification
+                                                  object:nil];
+  }
+}
+
+// Handles UIContentSizeCategoryDidChangeNotifications
+- (void)contentSizeCategoryDidChange:(NSNotification *)notification {
+  [self updateFontsForDynamicType];
+}
+
+// Update the fonts used based on mdc_preferredFontForMaterialTextStyle and recalculate the
+// preferred content size.
+- (void)updateFontsForDynamicType {
+  self.titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleTitle];
```

Comment:
ok let me rephrase: What happens if a client sets the Typogarphy's fontLoader to the MDFRobotoFontLoader and then also turns this bool on and ask for a prefered font? They would get system font. I would expect to get a Roboto font.
This is a bug.